src/mumble/Messages.cpp

```diff
@@ -970,20 +971,15 @@ void MainWindow::msgChannelState(const MumbleProto::ChannelState &msg) {
 	}
 
 	if (updateUI) {
-		// Passing nullptr to this function will make it do not much except fire a dataChanged event
-		// which leads to the UI being updated (reflecting the changes that just took effect).
-		this->pmModel->toggleChannelFiltered(nullptr);
+		this->pmModel->forceVisualUpdate();
 	}
 }
 
 void MainWindow::msgChannelRemove(const MumbleProto::ChannelRemove &msg) {
 	Channel *c = Channel::get(msg.channel_id());
 	if (c && (c->iId != 0)) {
-		if (c->bFiltered) {
-			ServerHandlerPtr sh = Global::get().sh;
-			if (sh)
-				Global::get().db->setChannelFiltered(sh->qbaDigest, c->iId, false);
-			c->bFiltered = false;
+		if (c->m_filterMode != Channel::FILTER_MODE_NORMAL) {
+			c->setFilterMode(Channel::FILTER_MODE_NORMAL);
```

Comment:
I think semantically clearer would be to have a small wrapper function `c->clearFilterMode` that essentially does
```cpp
if (c->m_filterMode != Channel::FILTER_MODE_NORMAL) {
    c->setFilterMode(Channel::FILTER_MODE_NORMAL);
}
```